### PR TITLE
Genesis block validations (hash etc.)

### PR DESCRIPTION
**Pull request by i1t**
Genesis block validations (hash etc.). Tests adjustments. Misc. fixes.

### CODE REVIEW

**Comment by i1t** [comment]
@macgregordennis @malroc 

It didn't seem valid that the hash isn't verified for genesis block and `Blockchain.load_genesis_block/0` was creating a block with invalid hash (I believe we shouldn't start the blockchain with such block).

Adjusted tests to match fixed hash for genesis block.

File: lib/credo_core_node/adapters/mining_adapter.ex [hunk 1]
```diff
@@ -43,7 +43,7 @@ defmodule CredoCoreNode.Adapters.MiningAdapter do
   @callback timelock_has_expired?(%Deposit{}, %PendingBlock{} | %Block{}) :: boolean()
   @callback relative_block_timelock_has_expired?(String.t(), %Block{}, %Deposit{}) :: boolean()
   @callback absolute_block_timelock_has_expired?(String.t(), %Block{}) :: boolean()
-  @callback timestamp_timelock_has_expired?(String.t(), %Block{}, %Deposit{} | nil) :: boolean()
+  @callback timestamp_timelock_has_expired?(String.t(), %Block{}) :: boolean()
```
Comment (i1t) [inline lib/credo_core_node/adapters/mining_adapter.ex]: A side fix to `master` implementation.

File: lib/credo_core_node/blockchain/blockchain.ex [hunk 2]
```diff
@@ -378,6 +378,38 @@ defmodule CredoCoreNode.Blockchain do
     write_and_dispatch_transactions(block)
   end
 
+  @doc """
+  Loads or creates genesis block with the following transactions:
```
Comment (i1t) [inline lib/credo_core_node/blockchain/blockchain.ex]: I believe we should annotate the functions with decoded `body` values (assuming no one can read the bytecode :-)).

**Comment by macgregordennis** [comment]
LGTM (looking through merged PRs)

**Pull request merged**